elip-21mil.mediawiki

```diff
@@ -0,0 +1,99 @@
+  License: BSD-3-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes a change to Elements to allow the unblinded issuance and reissuance of assets in amounts greater than the <code>MAX_MONEY</code> parameter (21 million) that is currently prevented. This is a hard-forking change, but can be safely released when combined with a new default policy config option that can prevent unblinded issuance transactions with amoutns greater than <code>MAX_MONEY</code> from being relayed and confirmed in blocks. 
+
+===Copyright===
+
+This document is licensed under the 3-clause BSD license.
+
+===Motivation===
+
+An important feature of the Elements platform is Confidential Assets, that enable any user to create tokens that represent external assets that can be transacted with strong privacy guarantees. Assets can be created in a special issuance transaction which also (optionally) creates a <code>reissuance_token</code> that enables additional issuances of the same original asset type. Both initial asset issuances and reissuances can be either blinded or unblinded, irrespective of whether further transfers of the asset are confidential. In a fully blinded issuance, the value (amount) of the issuance (and the reissuance token) remains hidden, however in an unblinded issuance (and reissuance) the amount of asset created is publicly verifiable. There are many situations where an unblinded issuance is needed, including for example transparency and legal requirements. 
+
+In the existing Elements consensus rules, in the function <code>VerifyIssuanceAmount</code> the value of an asset created is required to satisfy <code>MoneyRange(value)</code> (i.e. between zero and <code>MAX_MONEY</code>) if the value is explicit (unblinded). This check should only be required to apply to the <code>pegged_asset</code> which can never exceed <code>MAX_MONEY</code> (21 million bitcoin). To enable unblinded asset issuances greater than <code>MAX_MONEY</code> (which may be required for any non-bitcoin asset) this consensus rule needs to be changed to only apply the <code>MoneyRange(value)</code> check only to the <code>pegged_asset</code>. 
+
+==Design==
+
+===Overview===
+
+To enable the unblinded issuance and reissuance of assets other than the <code>pegged_asset</code> with values greater than <code>MAX_MONEY</code> (21 million) the <code>MoneyRange</code> check for explicit values in the <code>VerifyIssuanceAmount</code> consensus rule will be modified to only apply when the <code>asset</code> is <code>pegged_asset</code>. 
+
+To enable the inclusion of this consensus rule change in Elements releases without risking a network fork, a new policy rule preventing the unblinded issuance and reissuance of assets (and reissuance tokens) with values greater than <code>MAX_MONEY</code> before the the technical consensus hard fork is activated. This policy rule will be configurable but applied as default in all releases until the hard fork activation. 
```

Comment:
In f35d8c66c363277ae2d444d845e1db0a39d687b5:

This is a confusing -- if the hardfork isn't activated then the policy rule is meaningless. I think by "the hardfork activation" you mean "the time when the policy rule is disabled" but that's tautological.